### PR TITLE
OHRM5X-1947: Develop Claim - Create Claim Request API

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php [hunk 1]
```diff
@@ -118,32 +140,29 @@ private function setClaimRequest(ClaimRequest $claimRequest): ClaimRequest
...
             );
 
-            if ($this->getClaimService()->getClaimEventDao()->getClaimEventById($claimEventId) === null) {
+            $claimEvent = $this->getClaimService()->getClaimDao()->getClaimEventById($claimEventId);
+            if ($claimEvent === null) {
                 throw $this->getInvalidParamException(self::PARAMETER_CLAIM_EVENT_ID);
             }
 
             if ($this->getPayGradeService()->getPayGradeDao()->getCurrencyById($currencyId) === null) {
                 throw $this->getInvalidParamException(self::PARAMETER_CURRENCY);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php:149]:
```php
                throw $this->getInvalidParamException(self::PARAMETER_CURRENCY);
```
`PARAMETER_CURRENCY_ID` or `PARAMETER_CURRENCY_CODE`?

File: src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php [hunk 2]
```diff
@@ -118,32 +140,29 @@ private function setClaimRequest(ClaimRequest $claimRequest): ClaimRequest
...
             }
 
-            $claimRequest->setClaimEvent(
-                $this->getClaimService()->getClaimEventDao()->getClaimEventById($claimEventId)
-            );
-            $claimRequest->setCurrency($currencyId);
+            $claimRequest->setClaimEvent($claimEvent);
+            $claimRequest->setCurrencyType($this->getPayGradeService()->getPayGradeDao()->getCurrencyById($currencyId));
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php:153]: refactor using decorator

File: src/plugins/orangehrmClaimPlugin/Api/Model/ClaimExpenseTypeModel.php [hunk 3]
```diff
@@ -23,10 +23,31 @@
 use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
 use OrangeHRM\Entity\ExpenseType;
 
-class ClaimExpenseTypesModel implements Normalizable
+/**
+ * @OA\Schema(
+ *     schema="Claim-ExpenseTypesModel",
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/Model/ClaimExpenseTypeModel.php:28]:
```suggestion
 *     schema="Claim-ExpenseTypeModel",
```

File: src/plugins/orangehrmClaimPlugin/Api/Model/ClaimExpenseTypeModel.php [hunk 4]
```diff
@@ -23,10 +23,31 @@
...
+ *     ),
+ *     @OA\Property(
+ *         property="status",
+ *         type="boolean",
+ *     )
+ * )
+ */
+class ClaimExpenseTypeModel implements Normalizable
 {
     use ModelTrait;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/Model/ClaimExpenseTypeModel.php:50]: keep a space between class members

File: src/plugins/orangehrmClaimPlugin/Api/Model/ClaimRequestModel.php [hunk 5]
```diff
@@ -47,7 +70,8 @@ public function __construct(ClaimRequest $claimRequest)
                 'referenceId',
                 ['claimEvent', 'id'],
                 ['claimEvent', 'name'],
-                'currency',
+                ['currencyType', 'id'],
+                ['currencyType', 'name'],
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/Model/ClaimRequestModel.php:74]: hope this is align with `src/plugins/orangehrmPimPlugin/Api/Model/EmployeeSalaryModel.php`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/Migration/V5_4_0/Migration.php [hunk 6]
```diff
@@ -135,19 +116,156 @@ public function up(): void
...
+                    ->setParameter(
+                        'status',
+                        $this->setStatusToBoolean($row['status']),
+                        Types::BOOLEAN
+                    )
+                    ->executeStatement();
+            }
+            $result->free();
+        }
+    }
```
Comment (ManulMax) [inline installer/Migration/V5_4_0/Migration.php:239]: @RajithaKumara need to check whether this is ok or not.

File: src/plugins/orangehrmClaimPlugin/Api/Model/ClaimRequestModel.php (hunk 5 above)
Comment (ManulMax) [reply src/plugins/orangehrmClaimPlugin/Api/Model/ClaimRequestModel.php:74]: aligned with it. response come like this.
```
"currencyType": {
            "id": "USD",
            "name": "United States Dollar"
        },
```

File: src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php (hunk 1 above)
Comment (ManulMax) [reply src/plugins/orangehrmClaimPlugin/Api/ClaimRequestAPI.php]: add `PARAMETER_CURRENCY_ID`

File: installer/Migration/V5_4_0/Migration.php (hunk 6 above)
Comment (RajithaKumara) [reply installer/Migration/V5_4_0/Migration.php:239]: two update queries are enough right?

1. on => 1
2. null => 0

Comment (ManulMax) [reply installer/Migration/V5_4_0/Migration.php:239]: ya. I check there was no 'off' kind of thing. only 'on' and null

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**